hyper/http20/response.py

```diff
@@ -71,12 +77,8 @@ def __init__(self, headers, stream):
         # This 16 + MAX_WBITS nonsense is to force gzip. See this
         # Stack Overflow answer for more:
         # http://stackoverflow.com/a/2695466/1401686
-        if b'gzip' in self.headers.get(b'content-encoding', []):
-            self._decompressobj = zlib.decompressobj(16 + zlib.MAX_WBITS)
-        elif b'deflate' in self.headers.get(b'content-encoding', []):
-            self._decompressobj = DeflateDecoder()
-        else:
-            self._decompressobj = None
+        for c in self.headers.get(b'content-encoding', []):
+            self._decompressobj = decompressors.get(c)
```

Comment:
Nope, this is overbroad: we now `break` after the *first* attempt to get a decompressor, even if the content-encoding used is not a compression one. We need to check that `self._decompressobj is not None` before breaking.